### PR TITLE
feat: [CDS-88757]: Custom approval step schema example

### PR DESCRIPTION
**Pull request by sarthakkasat**

### CODE REVIEW

File: v1/examples/pipeline/steps/custom-approval-step.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      id: custom
+      spec:
+        steps:
+          - type: custom-approval
+            spec:
+              approval_criteria: 
+                expression: <+output.nameVal>=="123"
+              timeout: 10m
+              retry: 1m
+              shell: Bash
```
Comment (shalini-agr) [inline v1/examples/pipeline/steps/custom-approval-step.yaml:16]: this should be bash

File: v1/examples/pipeline/steps/custom-approval-step.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+              source:
+                type: inline
+                spec:
+                  script: |
+                    name=123
+                env_vars: []
+                out_vars: 
+                  - key: nameVal
+                    type: String
```
Comment (shalini-agr) [inline v1/examples/pipeline/steps/custom-approval-step.yaml:25]: this should be string

File: v1/examples/pipeline/steps/custom-approval-step.yaml (hunk 1 above)
Comment (sarthakkasat) [reply v1/examples/pipeline/steps/custom-approval-step.yaml]: Done

File: v1/examples/pipeline/steps/custom-approval-step.yaml (hunk 2 above)
Comment (sarthakkasat) [reply v1/examples/pipeline/steps/custom-approval-step.yaml]: Done

**Review by shalini-agr — APPROVED** [review]

File: v1/examples/pipeline/steps/custom-approval-step.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+spec:
+  stages: 
+    - type: custom
+      name: custom
+      id: custom
+      spec:
+        steps:
+          - type: custom-approval
+            spec:
+              approval_criteria:
```
Comment (SudarshanPurohit) [inline v1/examples/pipeline/steps/custom-approval-step.yaml:12]: Can we shorten this? Just "criteria", for example.

**Review by archit-harness — DISMISSED** [review]

File: v1/examples/pipeline/steps/custom-approval-step.yaml (hunk 3 above)
Comment (sarthakkasat) [reply v1/examples/pipeline/steps/custom-approval-step.yaml:12]: We have a rejection_criteria as well, hence we cannot name it to criteria

**Review by archit-harness — DISMISSED** [review]

**Review by SudarshanPurohit — APPROVED** [review]

**Review by archit-harness — APPROVED** [review]

**Pull request merged**